docs: improve dev containers documentation for user start-up (#15458)
coder/coder: docs/admin/index.md

For any information not strictly contained in these sections, check out our
[Tutorials](../tutorials/index.md) and [FAQs](../tutorials/faqs.md).

> ## What is an image, template, devcontainer, or workspace

Reviewer: This may be best presented as a table as this renders quite awkward.

Also I am not sure if this is best on the frontpage of administration. We have no other content here at all. If we do think this is the most important admin concept, we might want to add a sentence explaining why its so important and what else people can expect on the Administration page

<img width="748" alt="Screenshot 2024-11-24 at 9 10 03 PM" src="https://github.com/user-attachments/assets/df82f803-4167-4a3c-a95e-47e0fa3fecd5">

Author: I can try a table on and see how it feels - I'd tried an html definition list and it looked okay, but everything within the html would also have to be html (like links), and that didn't seem worth it. my concern with tables is how long text shows in cells, but it's worth a shot

the hope is to have this be a glossary component at some point so that we can define terms in place and have them combined into reference pages

here's an [example in the MyST docs](https://mystmd.org/guide/glossaries-and-terms)

Author: it's also interesting that you're still getting served a build of an older commit :-/ not sure why that is

Author: adjusted to:

![2024-11-26_14-45-53](https://github.com/user-attachments/assets/71e978e9-7877-47c6-aba6-180a88676d03)

- **Image**

  - A [base image](./templates/managing-templates/image-management.md) contains
    OS-level packages, and utilities that the Coder workspace is built on. It

>     can be an [example image](https://github.com/coder/images), custom image, or
>     one from [Docker Hub](https://hub.docker.com/search). It is defined in each

Reviewer: ```suggestion
    can be an [example image](https://github.com/coder/images), custom image in your registry, or
    one from [Docker Hub](https://hub.docker.com/search). It is defined in each
```

    for the workspace. For example, a template can include Kubernetes
    PersistentVolumeClaims, Docker containers, or EC2 VMs.
  - Managed by: Template administrators from within the Coder deployment.

- **Startup scripts**

  - Agent startup scripts apply to all users of a template. This is an
    intentionally flexible area that template authors have at their disposal to
    manage the "last mile" of workspace creation.
  - Managed by: Coder template administrators.

- **Workspace**

  - A [workspace](../user-guides/workspace-management.md) is the environment
    that a developer works in. Developers on a team each work from their own
    workspace and can use
    [multiple IDEs](../user-guides/workspace-access/index.md).
  - Managed by: Developers

- **Development containers**

  - A
    [development container](./templates/managing-templates/devcontainers/index.md)
    is an open-source specification for defining development environments. It is
    generally stored in VCS alongside associated source code. It can reference
    an existing base image, or a custom Dockerfile that will be built on-demand.

>   - Managed by: Developers

Reviewer: Suggest "Dev Containers" instead

Author: I updated your suggestion to do the dance around "Development Containers" as a specification and "dev containers" as the thing

> - **Development containers**
> 
>   - A
>     [development container](./templates/managing-templates/devcontainers/index.md)
>     is an open-source specification for defining development environments. It is
>     generally stored in VCS alongside associated source code. It can reference
>     an existing base image, or a custom Dockerfile that will be built on-demand.
>   - Managed by: Developers

Reviewer: ```suggestion
- **Development containers**

  - A
    [development container](./templates/managing-templates/devcontainers/index.md)
    is an open-source specification for defining development environments. It is
    generally stored in VCS alongside associated source code. It can reference
    an existing base image, or a custom Dockerfile that will be built on-demand.
  - Managed by: Dev Teams
```

- **Dotfiles / personalization**

  - Users may have their own specific preferences relating to shell prompt,
    custom keybindings, color schemes, and more. Users can leverage Coder's
    [dotfiles support](../user-guides/workspace-dotfiles.md) or create their own
    script to personalize their workspace. Be aware that users with root
    permissions in their workspace can override almost all of the previous
    configuration.

>   - Managed by: User

Reviewer: ```suggestion
  - Managed by: Individual Users
```
